web/packages/teleport/src/Discover/Shared/SetupAccess/useUserTraits.ts

```diff
@@ -107,8 +115,11 @@ export function useUserTraits(props: AgentStepProps) {
 
   // onProceed deduplicates and removes static traits from the list of traits
   // before updating user in the backend.
-  function onProceed(traitOpts: Partial<Record<Trait, Option[]>>) {
-    switch (props.resourceSpec.kind) {
+  function onProceed(
+    traitOpts: Partial<Record<Trait, Option[]>>,
```

Comment:
Is `Partial` needed here?

Reply:
yeah for each case we are partially (depending on resource) updating the fields defined here:
```ts
// UserTraits contain fields that define traits for local accounts.
export interface UserTraits {
  // logins is the list of logins that this user is allowed to
  // start SSH sessions with.
  logins: string[];
  // databaseUsers is the list of db usernames that this user is
  // allowed to open db connections as.
  databaseUsers: string[];
  // databaseNames is the list of db names that this user can connect to.
  databaseNames: string[];
  // kubeUsers is the list of allowed kube logins.
  kubeUsers: string[];
  // kubeGroups is the list of allowed kube groups for a kube cluster.
  kubeGroups: string[];
  // windowsLogins is the list of logins that this user
  // is allowed to start desktop sessions.
  windowsLogins: string[];
  // awsRoleArns is a list of aws roles this user is allowed to assume.
  awsRoleArns: string[];
}
```